### PR TITLE
Gpu examples

### PR DESCRIPTION
**Pull request by liruipeng**
This PR adds `TEST_examples` to GPU regression tests.

### CODE REVIEW

File: src/examples/ex12.c [hunk 1]
```diff
@@ -71,7 +71,7 @@ int main (int argc, char *argv[])
    HYPRE_Init();
 
    /* Print GPU info */
-   HYPRE_PrintDeviceInfo();
+   /* HYPRE_PrintDeviceInfo(); */
```
Comment (rfalgout) [inline src/examples/ex12.c:74]: Why comment this out vs removing it altogether?

**Review by rfalgout — APPROVED** [review]
Looks good, Ruipeng.  Thanks!

File: src/examples/ex12.c (hunk 1 above)
Comment (liruipeng) [reply src/examples/ex12.c:74]: Wanted to remind users they can print out info of GPU that is used by that. But turning it on pollutes the output, which fails regression tests.

File: src/test/TEST_examples/gpu.saved.cuda [hunk 2]
```diff
@@ -0,0 +1,72 @@
+# Output file: gpu.out.1
```
Comment (liruipeng) [inline src/test/TEST_examples/gpu.saved.cuda:1]: @rfalgout Do I name this file correctly (.saved.cuda) for `machine-lassen`? Not sure if the `save` mechanism of this test (with `-eo`) is treated the same as the others (with `-ro`). Thanks!

File: src/examples/ex12.c (hunk 1 above)
Comment (rfalgout) [reply src/examples/ex12.c:74]: Works for me.  Thanks!

File: src/test/TEST_examples/gpu.saved.cuda (hunk 2 above)
Comment (rfalgout) [reply src/test/TEST_examples/gpu.saved.cuda:1]: The mechanism is the same for '-ro:' and '-eo:'.  You need to use the '-rt' flag to pass additional options through to the `runtest.sh` script.  So, it looks like you need `-eo: -gpu -rt -mpibind -save cuda`.

File: AUTOTEST/machine-lassen.sh [hunk 3]
```diff
@@ -46,7 +46,7 @@ eo=""
 # CUDA with UM in debug mode [ij, ams, struct, sstruct]
 co="--with-cuda --enable-unified-memory --enable-persistent --enable-debug --with-cuda-arch=\\'60 70\\' --with-extra-CFLAGS=\\'-qmaxmem=-1 -qsuppress=1500-029\\' --with-extra-CXXFLAGS=\\'-qmaxmem=-1 -qsuppress=1500-029\\'"
 ro="-ij-gpu -ams -struct -sstruct -rt -mpibind -save cuda"
-./test.sh basic.sh $src_dir -co: $co -mo: $mo -ro: $ro
+./test.sh basic.sh $src_dir -co: $co -mo: $mo -ro: $ro -eo: -gpu
```
Comment (liruipeng) [inline AUTOTEST/machine-lassen.sh]: @rfalgout Just to confirm I need to have `-rt -mpibind -save cuda` in `-eo` even if I have them in `-ro`? Thanks again!

Comment (rfalgout) [reply AUTOTEST/machine-lassen.sh]: Yes.  The `-co:`, `-mo:`, `-ro:`, and `-eo:` options for `basic.sh` are completely separate from each other.  The information to put in each of them is passed directly to the `configure.sh`, `make.sh`, `run.sh`, and `examples.sh` tests.

Comment (liruipeng) [reply AUTOTEST/machine-lassen.sh]: Great! Thanks for explaining. I will make the change.

**Pull request merged**